src/logicTemplates/split.js

```diff
@@ -0,0 +1,121 @@
+        let injectedBytes = templates.inject(referenceProgramBytes, referenceOffsets, injectionVector, injectionTypes);
+        this.programBytes = injectedBytes;
+        this.address = templates.addressFromProgram(injectedBytes);
+        this.ratn = ratn;
+        this.ratd = ratd;
+        this.receiverOne = receiverOne;
+        this.receiverTwo = receiverTwo;
+    }
+
+    /**
+     * returns the program bytes
+     * @returns {Uint8Array}
+     */
+     getProgram() {
+        return this.programBytes;
+    }
+
+    /**
+     * returns the string address of the contract
+     * @returns {string}
+     */
+    getAddress() {
+        return this.address;
+    }
+
+    /**
+     * returns a group transactions array which transfer funds according to the contract's ratio
+     * @param {int} amount: the amount to be transferred
+     * @param {int} firstRound: the first round on which the transaction group will be valid
+     * @param {int} lastRound: the last round on which the transaction group will be valid
+     * @param {int} fee: the fee to pay in microAlgos
+     * @param {string} genesisHash: the b64-encoded genesis hash indicating the network for this transaction
+     * @param {boolean} precise, optional, precise treats the case where amount is not perfectly divisible based on the ratio.
+     *  When set to False, the amount will be divided as close as possible but one address will get
+     *  slightly more. When True, an error will be raised.
+     * @returns {Uint8Array}
+     */
+    getSendFundsTransaction(amount, firstRound, lastRound, fee, genesisHash, precise=true) {
+        let amountForReceiverOne = 0;
+        // reduce fractions
+        var gcdFn = function(a, b) {
+            if (!b) {
+                return a;
+            }
+
+            return gcdFn(b, a % b);
+        };
+        let gcd = gcdFn(this.ratn, this.ratd);
+        let ratn = Math.floor(this.ratn / gcd);
+        let ratd = Math.floor(this.ratd / gcd);
+        if (amount % ratd === 0) {
+            amountForReceiverOne = Math.floor(amount * ratn / ratd);
+        } else if (precise) {
+            throw Error("precise splitting requested but amount and contract ratio cannot be split precisely");
+        } else {
+            amountForReceiverOne = Math.round(amount * ratn / ratd);
+        }
+        let amountForReceiverTwo = amount - amountForReceiverOne;
+
+        let from = this.address;
+
+        let tx1 = algosdk.makePaymentTxn(from, this.receiverOne, fee, amountForReceiverOne, firstRound, lastRound, undefined, genesisHash, undefined, undefined);
+        let tx2 = algosdk.makePaymentTxn(from, this.receiverTwo, fee, amountForReceiverTwo, firstRound, lastRound, undefined, genesisHash, undefined, undefined);
```

Comment:
I think that these two lines don't take into account the change to the interface (no lease)